index.js

```diff
@@ -1,16 +1,20 @@
 const isObj = require('is-obj');
 
 
-const deepEqSkip = function(data, expectedData, skipKeys = []) {
+const deepEqSkip = function (data, expectedData, skipKeys = []) {
 
   // Validate data type of both the data
-  if (typeof(data) !== typeof(expectedData)) {
-    return false;
+  if (typeof (data) !== typeof (expectedData)) {
+    throw new Error(`Expected ${JSON.stringify(expectedData)} found ${JSON.stringify(data)}`);
   }
 
   // If data is not an object/array then match the values
   if (!isObj(data)) {
-    return (data === expectedData);
+    if (data !== expectedData) {
+      throw new Error(`Expected ${JSON.stringify(expectedData)} found ${JSON.stringify(data)}`);
+    }
+
+    return true;
```

Comment:
We should not return true from here.
As whenever the input data is object/array this will return true from here without actually matching it.

We should also have a test case for this, so we can capture these scenarios.
Can remove this and also write a test for this, please do run the test before and after removing to verify the tests.

Reply:
Not sure I understand what you mean.
_As whenever the input data is object/array this will return true from here without actually matching it_ can you give me an example?

Reply:
@radulucut My bad. I thought this condition `if (!isObj(data)) {` is `if (isObj(data)) {`

Its looks fine. Thanks. Sorry about the confusion